### ISSUE TITLE
Talk posts going to wrong project

### ISSUE THREAD

**Issue by camallen** [zooniverse/Panoptes-Front-End#5357]
## Expected behavior
With a subject that has been re-used from another project. When I start a new subject discussion on Talk, the UI should propose choices for the current project boards and not the subject project boards.

This only affects projects that are reusing subjects from a previous project.

## Current behavior
The new discussion boards in the UI are populated from the subject project and not the current project I am working on.

https://github.com/zooniverse/Panoptes-Front-End/blob/67f8c6fe4cac20c436ad56eced681f60d247ebc5/app/talk/discussion-new-form.cjsx#L36 

## Steps to replicate

1. Visit a subject's talk page e.g. https://www.zooniverse.org/projects/pangorilla/wild-gabon/talk/subjects/12048508
0. Click on 'start a new discussion'
0. The boards are populated from the subject project link, not the ignored project prop

https://github.com/zooniverse/Panoptes-Front-End/blob/21cf42485929a62938112d5e2d4bca1a6702b00e/app/subjects/comment-form.cjsx#L68

## Possible Solution
Allow the `discussion-new-form` component to have a project prop and ensure it's passed in and used instead of proxying from the subject's project.

## Additional information

- **Operating system:** ALL
- **Browser:** ALL

**Comment by eatyourgreens** [comment]
NB. this is a bug that is affecting Talk discussions on a live project: https://www.zooniverse.org/projects/pangorilla/wild-gabon/talk/2316/969017

**Comment by AnLand** [comment]
I don't necessarily want to open this issue again, but when I classify a subject and add it to my favorites or any other collection I create (in Wild Gabon) and open the subject then from this collection - it will always open the subject in the (finished) Elephant Expedition project and can only be posted on the latter's Talk boards.

Is it possible to do anything about this as well? There have been still Talk posts in the wrong project after the original issue here has been solved. I suspected the collections triggering part of the problem, but had only now time to test it.

Thanks a lot for looking into this again!
